services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/team/MembershipInvitationManagerImpl.java

```diff
@@ -162,6 +192,49 @@ public Count getOpenInvitationCountForUser(String principalId) {
 		return result;
 	}
 
+	@Override
+	public InviteeVerificationSignedToken verifyInvitee(Long userId, String membershipInvitationId) {
+		ValidateArgument.required(userId, "userId");
+		ValidateArgument.required(membershipInvitationId, "membershipInvitationId");
+		// Get list of email addresses associated with userId
+		List<PrincipalAlias> aliases = principalAliasDAO.listPrincipalAliases(userId);
+		List<String> emails = new ArrayList<>();
+		for (PrincipalAlias alias : aliases) {
+			if (alias.getType() == AliasType.USER_EMAIL) {
+				emails.add(alias.getAlias());
+			}
+		}
+		// Get inviteeEmail from membershipInvitation
+		String inviteeEmail = membershipInvtnSubmissionDAO.getInviteeEmail(membershipInvitationId);
+		// If inviteeEmail is in the emails list, construct InviteeVerificationSignedToken and return it
+		if (emails.contains(inviteeEmail)) {
+			InviteeVerificationSignedToken token = new InviteeVerificationSignedToken();
+			token.setInviteeId(userId.toString());
+			token.setMembershipInvitationId(membershipInvitationId);
+			SignedTokenUtil.signToken(token);
+			return token;
+		}
+		// inviteeEmail is not associated with the user, return null to convey invitee verification failure
+		return null;
+	}
+
+	@WriteTransactionReadCommitted
+	@Override
+	public void updateInviteeId(Long userId, String misId, InviteeVerificationSignedToken token) {
+		AuthorizationStatus status = authorizationManager.canAccessMembershipInvitationSubmission(userId, misId, token, ACCESS_TYPE.UPDATE);
+		if (!status.getAuthorized()) {
+			throw new UnauthorizedException(status.getReason());
+		}
+		MembershipInvtnSubmission mis = membershipInvtnSubmissionDAO.get(misId);
```

Comment:
this should be a select for update to ensure that no other thread changes the record before 'updateInviteeid' is callsed, below.